src/webui/api/torrentscontroller.cpp

```diff
@@ -531,27 +566,14 @@ void TorrentsController::trackersAction()
     if (!torrent)
         throw APIError(APIErrorType::NotFound);
 
-    QJsonArray trackerList = getStickyTrackers(torrent);
+    QJsonArray trackersList = getStickyTrackers(torrent);
+    QJsonArray trackers = getTrackers(torrent);
 
-    for (const BitTorrent::TrackerEntryStatus &tracker : asConst(torrent->trackers()))
-    {
-        const bool isNotWorking = (tracker.state == BitTorrent::TrackerEndpointState::NotWorking)
-                || (tracker.state == BitTorrent::TrackerEndpointState::TrackerError)
-                || (tracker.state == BitTorrent::TrackerEndpointState::Unreachable);
-        trackerList << QJsonObject
-        {
-            {KEY_TRACKER_URL, tracker.url},
-            {KEY_TRACKER_TIER, tracker.tier},
-            {KEY_TRACKER_STATUS, static_cast<int>((isNotWorking ? BitTorrent::TrackerEndpointState::NotWorking : tracker.state))},
-            {KEY_TRACKER_MSG, tracker.message},
-            {KEY_TRACKER_PEERS_COUNT, tracker.numPeers},
-            {KEY_TRACKER_SEEDS_COUNT, tracker.numSeeds},
-            {KEY_TRACKER_LEECHES_COUNT, tracker.numLeeches},
-            {KEY_TRACKER_DOWNLOADED_COUNT, tracker.numDownloaded}
-        };
-    }
+    // merge QJsonArray
+    for (auto &&tracker : trackers)
+        trackersList.append(std::move(tracker));
```

Comment:
To avoid detachment on `QJsonArray`:
```suggestion
    for (const auto &tracker : trackers)
        trackersList.append(tracker);
```

Reply:
Or even inline it:
```c++
for (const auto &tracker : asConst(getTrackers(torrent)))
    trackersList.append(tracker);
```

Reply:
That does seem better yes! Thanks for reviewing this, really appreciate it!